From foo@baz Mon Jan  5 15:44:13 PST 2009
Date: Mon, 05 Jan 2009 15:44:13 -0800
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: epl: hr timers all run in hard irq context now

From: Greg Kroah-Hartman <gregkh@suse.de>

Because of this, we can't set the mode for the timer, so delete this
code as it causes a build error right now.

Cc: Daniel Krueger <daniel.krueger@systec-electronic.com>
Cc: Ronald Sieber <Ronald.Sieber@systec-electronic.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/epl/TimerHighReskX86.c |    7 -------
 1 file changed, 7 deletions(-)

--- a/drivers/staging/epl/TimerHighReskX86.c
+++ b/drivers/staging/epl/TimerHighReskX86.c
@@ -225,13 +225,6 @@ tEplKernel PUBLIC EplTimerHighReskAddIns
 		hrtimer_init(pTimer, CLOCK_MONOTONIC, HRTIMER_MODE_ABS);
 
 		pTimer->function = EplTimerHighReskCallback;
-
-		/*
-		 * We use HRTIMER_CB_SOFTIRQ here.
-		 * HRTIMER_CB_IRQSAFE is critical as the callback function
-		 * would be called with IRQs disabled.
-		 */
-		pTimer->cb_mode = HRTIMER_CB_SOFTIRQ;
 	}
 
 	return Ret;
